### ISSUE TITLE
UI: Implement timeouts, retries for jobs

### ISSUE THREAD

**Issue by jweill-aws** [jupyter-server/jupyter-scheduler#169]
### Problem

This should be fixed after #90 (API) is done.

Users should be able to set timeouts and retry options for jobs.

### Proposed Solution

Implement these features in the create job UI, mapped to the fields listed below in the API.
- **timeout_seconds** timeout in seconds for the job to complete (must be a number)
- **retry_on_timeout** whether to retry if a timeout has occurred (default no)
- **max_retries** no of times to attempt retry (number)
- **min_retry_interval_millis** delay in milliseconds between retry attempts (number)

All of these inputs should be optional and should only be sent in the API request if they are present (nonblank/defined).

### Additional context

https://github.com/jupyter-server/jupyter-scheduler/pull/45#discussion_r986892626

**Comment by aiqc** [comment]
Agreed, especially on retries. Sometimes internet dependent processes get hung or a Docker container crashes for unknown reasons.
